crates/router/src/core/mandate.rs

```diff
@@ -259,6 +260,25 @@ where
     Ok(resp)
 }
 
+#[instrument(skip(state))]
+pub async fn retrieve_mandates_list(
+    state: &AppState,
+    merchant_account: storage::MerchantAccount,
+    constraints: api_models::mandates::MandateListConstraints,
+) -> RouterResponse<Vec<api_models::mandates::MandateResponse>> {
+    let mandates = state
+        .store
+        .find_mandates_by_merchant_id(&merchant_account.merchant_id, constraints)
+        .await
+        .to_not_found_response(errors::ApiErrorResponse::InternalServerError)
```

Comment:
```suggestion
        .change_context(errors::ApiErrorResponse::InternalServerError)
```

Reply:
Since we are searching unsing the merchant_id, shouldn't this error be `MerchantAccountNotFound`

Reply:
> Since we are searching unsing the merchant_id, shouldn't this error be `MerchantAccountNotFound`

Nope, we are returning a list of mandates, which would be empty when there are no results. See https://github.com/juspay/hyperswitch/pull/1143#discussion_r1192261008.